cmd/lotus-chainwatch/processor/processor.go

```diff
@@ -36,6 +36,9 @@ type Processor struct {
 
 	// number of blocks processed at a time
 	batch int
+
+	// process communication channels
+	sectorDealEvents chan *SectorDealEvent
```

Comment:
When the miner process iterates its precommits and finds one containing one or more deals the relevant info is sent on this channel. The market process consumes the info and persists its. This coupling between the miner and market process is required since the `minerid_dealid_sectorid` tables `deal_id` column references `deal_id` in `market_deal_proposals`.